http/src/jsMain/kotlin/com.mirego.trikot.http/web/WebHttpResponse.kt

```diff
@@ -0,0 +1,33 @@
+package com.mirego.trikot.http.web
+
+import com.mirego.trikot.http.HttpResponse
+import org.khronos.webgl.ArrayBuffer
+import org.khronos.webgl.Int8Array
+import org.w3c.xhr.XMLHttpRequest
+
+fun getResponseBody(xhr: XMLHttpRequest): ByteArray? {
+    return xhr.response?.let {
+        val response: ArrayBuffer = it as ArrayBuffer
+        val length = it.byteLength
+        Int8Array(response, 0, length).unsafeCast<ByteArray>()
+    }
+}
+
+fun getResponseHeaders(xhr: XMLHttpRequest): Map<String, String> {
+    return xhr.getAllResponseHeaders()
```

Comment:
I sure would love to see some tests especially for that part!

Reply:
Where are the tests for the Android and iOS version?

Reply:
👀 🙄  I think its more the splitting part that @npresseault was talking about rather than the `getAllResponseHeaders()`

Libraries are not written in the common code. I think this is the reason why.

Reply:
Yep, especially the header splitting / parsing. This would be easy to isolate and test if you really don't want to test the whole thing.

Although, I feel having tests for Android and iOS would be a nice addition too eventually. We should especially have some integration tests which could target Web / Android / Web to ensure all behaviors are the same.

I agree it is out of scope for your PR though so only testing this small bit would be a start !

Reply:
I must be missing a comment or something, because @npresseault did not talk about splitting anything.

But, if we are to talk about that, it seems that the gradle hooks to publish a kotlin library with JS compatibility on maven are only present when the kotlin-multiplatform plugin is used. So to split this code into it's own separate project, I would have to create an entirely new kotlin-multiplatform project with only the JS stuff in it. Wouldn't it be absurd, since we already have a KMP project with all the proper bindings to interop with JS in the jsMain target? I feel like KMP libraries are actually intended to be implemented this way.